flowmachine/flowmachine/core/query_state.py

```diff
@@ -0,0 +1,346 @@
+
+    def cancel(self):
+        """
+        Attempt to mark the query as cancelled.
+
+        Returns
+        -------
+        tuple of QueryState, bool
+            Returns a tuple of the new query state, and a bool indicating whether the caller
+            triggered the query to be cancelled with this call
+
+        """
+        return self.trigger_event(QueryEvent.CANCEL)
+
+    def enqueue(self):
+        """
+        Attempt to mark the query as queued.
+
+        Returns
+        -------
+        tuple of QueryState, bool
+            Returns a tuple of the new query state, and a bool indicating whether the caller
+            triggered the query to be queued with this call
+
+        """
+        return self.trigger_event(QueryEvent.QUEUE)
+
+    def raise_error(self):
+        """
+        Attempt to mark the query as having errored while running.
+
+        Returns
+        -------
+        tuple of QueryState, bool
+            Returns a tuple of the new query state, and a bool indicating whether the caller
+            marked the query as erroring with this call
+
+        """
+        return self.trigger_event(QueryEvent.ERROR)
+
+    def execute(self):
+        """
+        Attempt to mark the query as in the process of executing.
+
+        Returns
+        -------
+        tuple of QueryState, bool
+            Returns a tuple of the new query state, and a bool indicating whether the caller
+            marked the query as executing with this call
+
+        """
+        return self.trigger_event(QueryEvent.EXECUTE)
+
+    def finish(self):
+        """
+        Attempt to mark the query as completed.
+
+        Returns
+        -------
+        tuple of QueryState, bool
+            Returns a tuple of the new query state, and a bool indicating whether the caller
+            marked the query as finished with this call
+
+        """
+        return self.trigger_event(QueryEvent.FINISH)
+
+    def reset(self):
+        """
+        Attempt to mark the query as in the process of resetting.
+
+        Returns
+        -------
+        tuple of QueryState, bool
+            Returns a tuple of the new query state, and a bool indicating whether the caller
+            triggered the query to be reset with this call
+
+        """
+        return self.trigger_event(QueryEvent.RESET)
+
+    def finish_resetting(self):
+        """
+        Attempt to mark the query as having finished resetting.
+
+        Returns
+        -------
+        tuple of QueryState, bool
+            Returns a tuple of the new query state, and a bool indicating whether the caller
+           marked the reset as complete with this call.
+
+        """
+        return self.trigger_event(QueryEvent.FINISH_RESET)
+
+    def wait_until_complete(self):
+        """
+        Blocks while the query is in any state that makes how to get
+        the result of it indeterminate (i.e. currently running, resetting,
+        or scheduled to run).
+
+        Returns
```

Comment:
See comment above - it's a bit confusing (at least in Python land) that a function called `wait_until_complete()` has a return value. Can we remove this return value, since the user can just as easily check the `is_completed` state themselves, which is also clearer to read?